internal/charm/charm.go

```diff
@@ -25,25 +22,6 @@ type Charm interface {
 	Revision() int
 }
 
-// ReadCharm reads a Charm from path, which can point to either a charm archive or a
-// charm directory.
-func ReadCharm(path string) (charm Charm, err error) {
-	info, err := os.Stat(path)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if info.IsDir() {
-		charm, err = ReadCharmDir(path)
-	} else {
-		charm, err = ReadCharmArchive(path)
-	}
```

Comment:
As pointed out, we know that we should only ever read an archive. The uniter never reads back from the filesystem, and only places files on disk during a fresh charm install, or a charm refresh.

It's _very_ clear now that charms are archives.